src/test/java/com/firebolt/jdbc/client/query/StatementClientImplTest.java

```diff
@@ -107,6 +93,43 @@ private Entry<String, String> shouldPostSqlQuery(boolean systemEngine) throws SQ
 		return Map.entry(statementInfoWrapper.getLabel(), actualRequest.url().toString());
 	}
 
+	@ParameterizedTest
+	@CsvSource({
+			"true,2,queryLabelFromConnection",
+			"true,2,null"
+	})
+	void shouldPostSqlWithExpectedQueryLabel(boolean systemEngine, int infraVersion, String connectionQueryLabel) throws SQLException, IOException {
+		FireboltProperties fireboltProperties = FireboltProperties.builder().database("db1").compress(true).host("firebolt1").port(555).accountId("12345").systemEngine(systemEngine)
+						.runtimeAdditionalProperties(Map.of("query_label", connectionQueryLabel)).build();
+		when(connection.getAccessToken()).thenReturn(Optional.of("token"));
+		when(connection.getInfraVersion()).thenReturn(infraVersion);
+
+		StatementClient statementClient = new StatementClientImpl(okHttpClient, connection, "ConnA:1.0.9", "ConnB:2.0.9");
+		injectMockedResponse(okHttpClient, 200, "");
+		Call call = getMockedCallWithResponse(200, "");
+		when(okHttpClient.newCall(any())).thenReturn(call);
+		StatementInfoWrapper statementInfoWrapper = StatementUtil.parseToStatementInfoWrappers("show databases").get(0);
+		assertNotNull(statementInfoWrapper.getLabel());
+		statementClient.executeSqlStatement(statementInfoWrapper, fireboltProperties, fireboltProperties.isSystemEngine(), QUERY_TIMEOUT);
+
+		verify(okHttpClient).newCall(requestArgumentCaptor.capture());
+		Request actualRequest = requestArgumentCaptor.getValue();
+		Map<String, String> expectedHeaders = new LinkedHashMap<>();
+		expectedHeaders.put("Authorization", "Bearer token"); // legit:ignore
+		expectedHeaders.put("User-Agent", userAgent("ConnB/2.0.9 JDBC/%s (Java %s; %s %s; ) ConnA/1.0.9"));
+		assertEquals(expectedHeaders, extractHeadersMap(actualRequest));
```

Comment:
I'd remove this block as well - user agent is non-essential part of our code and I really would like to avoid having more tests than necessary to depend on it.